saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/algorithm/SavedSitesSyncPersisterAlgorithm.kt

```diff
@@ -58,7 +58,9 @@ class RealSavedSitesSyncPersisterAlgorithm @Inject constructor(
     override fun processEntries(
         bookmarks: SyncBookmarkEntries,
         conflictResolution: SyncConflictResolution,
-    ): SyncMergeResult<Boolean> {
+    ): SyncMergeResult {
+        var orphans = false
```

Comment:
We discussed this one over zoom:

Notes:
- To detect orphans we rely on `processIds` and `allResponseIds`.
- Any id from payload which is not present in `processIds` is an orphan.
- However our implementation is able to process all items on a payload.
- We defined an orphan as a savedsite (bokmark or folder) from payload, without a parent in the payload AND not present on the local store.
- For those cases, even if we apply our fallback logic (adding them as childs of root folder), we should not add their ids to `processIds`